Source/WebCore/platform/ios/ScrollAnimatorIOS.mm

```diff
@@ -88,7 +88,7 @@
         return false;
     }
     
-    IntPoint currentPoint = touchEvent.touchLocationAtIndex(0);
+    IntPoint currentPoint = flooredIntPoint(touchEvent.touchLocationAtIndex(0));
```

Comment:
It feels like all this logic should be float-based (being careful about floating point comparison).